sorts/mergesort_top-down.ts

```diff
@@ -0,0 +1,40 @@
+/* 
+Implementation of in-place top-down merge sort with stack.
+  complexity:
+    best time - O(N log N)
+    avg time - O(N log N)
+    worst time - O(N log N)
+    
+    memory - O(N)
+    stable - Yes
+*/
+
+function MergeArrays(arr1: number[], arr2: number[], target: number[]) {
+  let i, i1, i2;
+  i = i1 = i2 = 0;
+
+  while (i1 < arr1.length || i2 < arr2.length) {
+    if (i2 >= arr2.length || (i1 < arr1.length && arr1[i1] < arr2[i2])) {
+      target[i++] = arr1[i1++];
+    } else {
+      target[i++] = arr2[i2++];
+    }
+  }
+
+  if (target.length != arr1.length + arr2.length) {
```

Comment:
Could that happen?

Reply:
Actually never, but I'm paranoid so much sometimes, that I do that stuff. Ye, I am removing it now. Do you like the overall code?